pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -412,65 +479,16 @@ func (o *MustGatherOptions) Run() error {
 	defer o.logTimestamp()
 
 	var wg sync.WaitGroup
-	wg.Add(len(pods))
 	errCh := make(chan error, len(pods))
+	podChan := make(chan *corev1.Pod)
+	wg.Add(concurrentMG)
+	for i := 0; i < concurrentMG; i++ {
+		go o.runWorker(ns.Name, podChan, errCh, &wg)
+	}
 	for _, pod := range pods {
```

Comment:
Can you move
```
	for _, pod := range pods {
		podChan <- pod
	}
	close(podChan)
```
before running
```
	for i := 0; i < concurrentMG; i++ {
		go o.runWorker(ns.Name, podChan, errCh, &wg)
	}
```
? This will make the code easier to read.

Reply:
Sure

Reply:
Done.